lib/engine.js

```diff
@@ -1,31 +1,33 @@
 const panelLeague = require('./panel-league/stepper.js');
 const defaultTo = require('lodash/defaultTo');
 
-const STATE_CACHE_SIZE = 16;
+const STATE_CACHE_SIZE = 128;
 
 const defaultOptions = {
   width: 6,
   height: 8,
   stepper: "panelLeagueStandard",
+  stateCacheSize: STATE_CACHE_SIZE,
 };
 
 class GameEngine {
   constructor(options = {}) {
     const stepper = defaultTo(options.stepper, defaultOptions.stepper);
     this.stepper = panelLeague.factory(stepper);
+    this.stateCacheSize = defaultTo(options.stateCacheSize, defaultOptions.stateCacheSize);
     // Width and height are technically game specific but we cache them for convenience.
     const width = defaultTo(options.width, defaultOptions.width);
     const height = defaultTo(options.height, defaultOptions.height);
     const colors = options.colors;
     const state = this.stepper.initializeState(options);
 
-    this.time = state.time;
+    this._time = state.time;
     this.eventsByTime = {};
     this.initialState = JSON.stringify(state);
 
     // Smart state cache for handling input lag.
     this.statesByTime = new Map();
-    this.lastValidTime = this.time;
+    this.lastValidTime = null;
```

Comment:
Here `null` acts as a special flag to recalculate the state so `undefined` feels wrong to me. I could use a `Symbol()` if you like.